BigQueryWorkloadTester/src/main/java/com/google/cloud/pontem/benchmark/runners/ConcurrentWorkloadRunnerFactory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.pontem.benchmark.runners;
+
+import com.google.cloud.pontem.benchmark.backends.BigQueryBackend;
+import com.google.cloud.pontem.benchmark.runners.callables.WorkloadRunnerCallable;
+import com.google.cloud.pontem.config.WorkloadSettings;
+import com.google.cloud.pontem.model.WorkloadResult;
+import com.google.common.base.Stopwatch;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.Logger;
+
+/** A factory for ConcurrentWorkloadRunner objects. */
+public class ConcurrentWorkloadRunnerFactory {
+  private static final Logger logger =
+      Logger.getLogger(ConcurrentWorkloadRunnerFactory.class.getName());
+
+  // See: https://cloud.google.com/bigquery/quotas
+  private static final int CONCURRENCY_LIMIT = 50;
```

Comment:
Concurrency for what? queries? connections?
could we have a more verbose name?

Reply:
Done